doc/query_dsl.md

```diff
@@ -1,15 +1,416 @@
-# Commando Query DSL
+<div align="center">
+<h1>Query DSL</h1>
+</div>
 
-This mechanism positions itself as a built-in, lightweight, and somewhat limited alternative to GraphQL or Pathom3 (Pathom3 is a data query and transformation library for Clojure). Commando DSL is much more primitive and requires you to describe dependency resolution yourself. Let's look at how it works.
+The Commando Query DSL is a built-in, lightweight query mechanism. It serves as a simple alternative to more comprehensive solutions like GraphQL or Pathom3, but it is more primitive and requires you to define dependency resolution manually.
 
+Its primary purpose is to provide a way to:
 
-## Example Database Setup
+1. **Selectively query data, returning only the fields they request.**
+
+2. **Handle nested data dependencies through lazy-loading resolvers.**
+
+## Content
+
+- [Core Concept](#core-concept)
+- [Workaround](#workaround)
+- [Lazy Resolution](#lazy-resolution)
+  - [Lazy Resolution Examples](#lazy-resolution-examples) 
+- [Parameterization and Overriding](#parameterization-and-overriding)
+  - [Parametrization Examples](#parametrization-examples)
+- [Real-World Example](#real-world-example)
+  - [Examples Queries](#examples-queries)
+- [Advanced Topics](#advanced-topics)
+	- [Combining Mutations and Queries](#combining-mutations-and-queries)
+	- [Working with JSON](#working-with-json)
+- [Summary](#summary)	
+
+## Core Concept
+
+The Query DSL is enabled by adding `commando.commands.query-dsl/command-resolve-spec` to your `(commando.core/execute [<spec>] <instruction>)` call.
```

Comment:
"by adding command-resolve-spec to **commando execute registry**. "
To be more specific and align with commando wording, we just need to register it (add to registry).